tests/test_compile.py

```diff
@@ -0,0 +1,126 @@
+# Copyright 2025 The HuggingFace Team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import unittest
+
+import torch
+from torch.utils.benchmark import Timer
+
+from accelerate.test_utils import require_huggingface_suite, require_non_cpu, torch_device
+from accelerate.utils import compile_regions, extract_model_from_parallel, release_memory
+
+
+MODEL_ID = "gpt2"
+
+COMPILE_ITERS = 2
+INFERENCE_ITERS = 100
+
+INFRENCE_STMT = "model(input_ids, use_cache=False)"
+COMPILE_STMT = f"torch._dynamo.reset(); torch._inductor.utils.clear_inductor_caches(); {INFRENCE_STMT}"
+
+
+@require_huggingface_suite
+class RegionalCompilationTester(unittest.TestCase):
+    def _get_model_and_inputs(self):
+        from transformers import AutoConfig, AutoModelForCausalLM
+
+        with torch.device(torch_device):
+            config = AutoConfig.from_pretrained(MODEL_ID)
+            model = AutoModelForCausalLM.from_config(config)
+            input_ids = torch.randint(0, 1000, (4, 128), dtype=torch.int64)
+
+        return model, input_ids
+
+    def test_dynamo_extract_model_keep_torch_compile(self):
+        model, _ = self._get_model_and_inputs()
+        compiled_model = compile_regions(model)
+
+        # could also do a test with DistributedDataParallel, but difficult to run on CPU or single GPU
+        distributed_model = torch.nn.parallel.DataParallel(model)
+        distributed_compiled_model = compile_regions(distributed_model)
+        compiled_model_unwrapped = extract_model_from_parallel(distributed_compiled_model, keep_torch_compile=True)
+
+        assert compiled_model._orig_mod == compiled_model_unwrapped._orig_mod
+
+    def test_dynamo_extract_model_remove_torch_compile(self):
+        model, _ = self._get_model_and_inputs()
+        compiled_model = compile_regions(model)
+
+        # could also do a test with DistributedDataParallel, but difficult to run on CPU or single GPU
```

Comment:
Should the test be decorated with something that says it needs multiple GPUs? 👁️